### PR TITLE
feat(trace): add storage proof about l1fee (baseFee, overhead, scalar) and withdraw root into trace

### PR DESCRIPTION
**Pull request by noel2004**
This PR added proofs of zktrie into `storageProofs` of block trace. So circuit can prove the reading op from predeployed contract `L1GasPriceOracleAddress` about the l1 fee factors  (baseFee, overhead and scalar), and `	L2MessageQueueAddress` about the withdraw root 

~~The factors deriding l1fee for current transaction should be included in the block trace so circuit can verify the calculating. So we add all the required factors (baseFee, overhead and scalar) for calculating l1fee , which are storaged in `L1GasPriceOracleAddress` contract, into `ExecutionResult`. we also made efforts to [export these factors from `StateTransaction`](https://github.com/scroll-tech/go-ethereum/blob/6d8c2ebf762a1a932e1a9290ce62906cfbdc9080/core/state_transition.go#L74).~~

~~Moreover, we notice all of these reading of the storages from predeployed contracts require corresponding proofs of zktrie being added into `storageProofs` of block trace.~~

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1204547993466425

### CODE REVIEW

File: core/state_transition.go [hunk 1]
```diff
@@ -289,13 +299,13 @@ func (st *StateTransition) preCheck() error {
...
-// - concrete execution error:
-//      various **EVM** error which aborts the execution,
-//      e.g. ErrOutOfGas, ErrExecutionReverted
+//   - used gas:
+//     total gas used (including gas being refunded)
+//   - returndata:
+//     the returned data from evm
+//   - concrete execution error:
+//     various **EVM** error which aborts the execution,
+//     e.g. ErrOutOfGas, ErrExecutionReverted
```
Comment (0xmountaintop) [inline core/state_transition.go]: can you revert these formatting so that we can reduce the diff for auditing?

File: core/state_transition.go [hunk 2]
```diff
@@ -44,8 +49,10 @@ The state transitioning model does all the necessary work to work out a valid ne
 3) Create a new state object if the recipient is \0*32
 4) Value transfer
 == If contract creation ==
-  4a) Attempt to run transaction data
-  4b) If valid, use result as code for the new state object
+
+	4a) Attempt to run transaction data
+	4b) If valid, use result as code for the new state object
+
```
Comment (0xmountaintop) [inline core/state_transition.go]: ditto

File: eth/tracers/api_blocktrace.go [hunk 3]
```diff
@@ -314,6 +316,12 @@ func (api *API) getTxResult(env *traceEnv, state *state.StateDB, index int, bloc
 	}
 
 	proofStorages := tracer.UpdatedStorages()
+	proofStorages[rcfg.L1GasPriceOracleAddress] = vm.Storage(
+		map[common.Hash]common.Hash{
+			rcfg.L1BaseFeeSlot: {},
```
Comment (Thegaram) [inline eth/tracers/api_blocktrace.go]: Won't using `{}` here mean that we'll have `isDelete = true` below?

File: eth/tracers/api_blocktrace.go [hunk 4]
```diff
@@ -404,6 +412,35 @@ func (api *API) fillBlockTrace(env *traceEnv, block *types.Block) (*types.BlockT
 		txs[i] = types.NewTransactionData(tx, block.NumberU64(), api.backend.ChainConfig())
 	}
 
+	if _, existed := env.Proofs[rcfg.L2MessageQueueAddress.String()]; !existed {
```
Comment (Thegaram) [inline eth/tracers/api_blocktrace.go:415]: Why are `L1GasPriceOracleAddress` and `L2MessageQueueAddress` handled at different parts of the code?

**Review by Thegaram — DISMISSED** [review]

File: eth/tracers/api_blocktrace.go [hunk 5]
```diff
@@ -404,6 +412,35 @@ func (api *API) fillBlockTrace(env *traceEnv, block *types.Block) (*types.BlockT
...
+				wrappedProof[i] = bt
+			}
+			env.Proofs[rcfg.L2MessageQueueAddress.String()] = wrappedProof
+		}
+	}
+
+	if _, existed := env.StorageProofs[rcfg.L2MessageQueueAddress.String()]; !existed {
+		env.StorageProofs[rcfg.L2MessageQueueAddress.String()] = make(map[string][]hexutil.Bytes)
+	}
+	if _, existed := env.StorageProofs[rcfg.L2MessageQueueAddress.String()][rcfg.WithdrawTrieRootSlot.String()]; !existed {
```
Comment (colinlyguo) [inline eth/tracers/api_blocktrace.go:430]: Would this bloat?

File: eth/tracers/api_blocktrace.go (hunk 3 above)
Comment (noel2004) [reply eth/tracers/api_blocktrace.go]: Not that. The code below would only access the storage key so it is not matter here what the storage value is.

I would add a comment about that.

File: eth/tracers/api_blocktrace.go (hunk 4 above)
Comment (noel2004) [reply eth/tracers/api_blocktrace.go:415]: You are right, we can collect all of these proofs only one time. And itt maybe better to put the proofs at the beginning of a block, considering the 'block sealing' API would be induced later.

**Review by colinlyguo — DISMISSED** [review]

File: eth/tracers/api_blocktrace.go (hunk 4 above)
Comment (colinlyguo) [inline eth/tracers/api_blocktrace.go:415]: Under what circumstances would the key `rcfg.L2MessageQueueAddress.String()` exist in the `env.Proofs` map, thus making existed true?

Comment (Thegaram) [reply eth/tracers/api_blocktrace.go:415]: For example when `L2Messenger` appends a message to `L2MessageQueue`.

Comment (Thegaram) [reply eth/tracers/api_blocktrace.go:415]: We can refactor this later.

**Review by Thegaram — DISMISSED** [review]

**Review by colinlyguo — DISMISSED** [review]

**Review by 0xmountaintop — APPROVED** [review]

**Pull request merged**